why3/src/declaration.rs

```diff
@@ -143,6 +143,7 @@ pub enum Attribute {
 #[cfg_attr(feature = "serialize", derive(Serialize, Deserialize))]
 pub struct Signature {
     pub name: Ident,
+    pub trigger: Option<Trigger>, // None means Trigger::single(function_call(self))
```

Comment:
Is the comment really true? That seems like an awfully strong guarantee about the meaning of inferred triggers.

Reply:
`Option::None` means that the flag --simple-triggers was passed in, the signature is not a `law`, and it doesn't return `()` so we should use `result` as per the description of --simple-triggers (just an optimization generate this expression lazily).

`Some(Trigger::NONE)` means we won't add a trigger and let the tool infer it.